src/time_steppers.jl

```diff
@@ -120,23 +119,15 @@ function store_previous_source_terms!(grid::Grid, Gu, Gv, Gw, GT, GS, Gpu, Gpv,
     @synchronize
 end
 
-@inline δρ(ρ₀, βT, T₀, T, i, j, k) = @inbounds -ρ₀ * βT * (T[i, j, k] - T₀)
-
 "Update the hydrostatic pressure perturbation pHY′ and buoyancy δρ."
-function update_buoyancy!(grid::Grid, constants, eos, δρ, T, pHY′)
-    ρ₀, T₀, βT = eos.ρ₀, eos.T₀, eos.βT
+function update_buoyancy!(grid::Grid, constants, eos, T, pHY′)
     gΔz = constants.g * grid.Δz
 
-    @loop for k in (1:grid.Nz; blockIdx().z)
-        @loop for j in (1:grid.Ny; (blockIdx().y - 1) * blockDim().y + threadIdx().y)
-            @loop for i in (1:grid.Nx; (blockIdx().x - 1) * blockDim().x + threadIdx().x)
-                @inbounds δρ[i, j, k] = -ρ₀ * βT * (T[i, j, k] - T₀)
-
-                ∫δρ = (-ρ₀*βT*(T[i, j, 1]-T₀))
-                for k′ in 2:k
-                    ∫δρ += ((-ρ₀*βT*(T[i, j, k′-1]-T₀)) + (-ρ₀*βT*(T[i, j, k′]-T₀)))
-                end
-                @inbounds pHY′[i, j, k] = 0.5 * gΔz * ∫δρ
+    @loop for j in (1:grid.Ny; (blockIdx().y - 1) * blockDim().y + threadIdx().y)
+        @loop for i in (1:grid.Nx; (blockIdx().x - 1) * blockDim().x + threadIdx().x)
+            @inbounds pHY′[i, j, 1] = 0.5 * gΔz * δρ(eos, T, i, j, 1)
+            for k in 2:grid.Nz
```

Comment:
You might want to use the `@unroll` macro from GPUifyLoops here. Is `grid.Nz` a compile time constant value?

Reply:
Ah yes, `grid.Nz` should be a compile time constant `::Int`. Didn't know about `@unroll`, will give it a try!